File: Strings/Easy/415/415.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        carry = 0
        result = []
        i, j = len(num1) - 1, len(num2) - 1

        while i >= 0 or j >= 0:
            x = int(num1[i]) if i >= 0 else 0
            y = int(num2[j]) if j >= 0 else 0
            carry, digit = divmod(x + y + carry, 10)
            result.append(str(digit))
            i, j = i - 1, j - 1

        if carry:
            result.append(str(carry))

        return ''.join(result[::-1])

File: Strings/Easy/415/415.c
char * addStrings(char * num1, char * num2){
    int len1 = strlen(num1);
    int len2 = strlen(num2);
    int maxLen = len1 > len2 ? len1 : len2;
    int carry = 0;

    char *res = (char *)malloc(maxLen + 2);

    int i = len1 - 1;
    int j = len2 - 1;
    int k = 0;

    while(i >= 0 || j >= 0 || carry){
        int n1 = i >= 0 ? num1[i] - '0' : 0;
        int n2 = j >= 0 ? num2[j] - '0' : 0;

        int sum = n1 + n2 + carry;
        carry = sum / 10;
        res[k] = (sum % 10) + '0';

        i--;
        j--;
        k++;
    }
    int end = k - 1;
    for(int start = 0; start < end; start++){
        char temp = res[start];
        res[start] = res[end];
        res[end--] = temp;
    }
    res[k] = '\0';
    return res;
}